ompi/communicator/communicator.h

```diff
@@ -561,7 +579,7 @@ static inline ompi_communicator_t *ompi_comm_lookup_cid (const ompi_comm_extende
 static inline struct ompi_proc_t* ompi_comm_peer_lookup (const ompi_communicator_t* comm, const int peer_id)
 {
 #if OPAL_ENABLE_DEBUG
-    if(peer_id >= comm->c_remote_group->grp_proc_count) {
+    if((NULL == comm->c_remote_group) || (peer_id >= comm->c_remote_group->grp_proc_count)) {
```

Comment:
Why would the remote_group be NULL on a valid communicator ?

Reply:
okay a little background.  HCOLL must be using the OMPI devel headers so is bypassing the usual parameter checks for whether a comm is valid and is going directly into ompi_comm_size etc.   so we have to put checks in this code since HCOLL decided to use ompi devel/internal headers.

Reply:
I understand but these checks are affecting everybody, when the real issue is in the HCOLL module. Why is not the fix specialized for the mca/coll/hcoll component ?